docs/user/guides/setup/configuration-file.rst

```diff
@@ -0,0 +1,223 @@
+
+As part of the initial setup for your Read the Docs site,
+you need to create a **configuration file** called ``.readthedocs.yaml``.
+The configuration file tells Read the Docs what specific settings to use for your project.
+
+This how-to guide covers:
+
+#. Where to put your configuration file.
+#. Templates to setup your configuration file from scratch.
+#. The most basic sections in the configuration file that can help you get started.
+#. Next steps...
+
+.. seealso::
+
+   :doc:`/tutorial/index`.
+     Following the steps in our tutorial will help you setup your first documentation project.
+
+
+Where to put your configuration file
+------------------------------------
+
+The ``.readthedocs.yaml`` file should be placed in the top-most directory of your project's repository.
+We will get to the contents of the file in the next steps.
+
+When you have changed the configuration file,
+you need to commit and push the changes to your Git repository.
+Read the Docs will then automatically find and use the configuration to build your project.
+
+.. note::
+
+    Why is it called .yaml? What is that?
+      The configuration file is a `YAML`_ file. YAML files are a "map": a collection of
+      key-value pairs that can be nested. This is not unlike a JSON file or ``dict``
+      object in Python. This page won't explain the structure of YAML files, but many resources exist
+      online.
+
+
+.. _YAML: https://en.wikipedia.org/wiki/YAML
+
+.. _howto_templates:
+
+Getting started with a template
+-------------------------------
+
+Here are some variations of the configuration file that can help you get started.
+Pick the one that resembles your project the most,
+copy its contents to ``.readthedocs.yaml`` and add the file to your Git repository.
+
+.. tabs::
+
+    .. tab:: Sphinx
+
+        If your project uses Sphinx,
+        we offer a special builder optimized for Sphinx projects.
+
+        .. literalinclude:: /config-file/examples/sphinx/.readthedocs.yaml
+           :language: yaml
+           :linenos:
+
+
+    .. tab:: MkDocs
+
+        If your project uses MkDocs,
+        we offer a special builder optimized for MkDocs projects.
+
+        .. literalinclude:: /config-file/examples/mkdocs/.readthedocs.yaml
+           :language: yaml
+           :linenos:
+
+    .. tab:: Other tools
+
+        If you are using another documentation tool,
+        you can configure the commands of your documentation tool in the following code.
+
+        You will need to create the output directories and direct your documentation tool to write its outputs into those directories.
+
+        .. literalinclude:: /config-file/examples/custom-commands-echo/.readthedocs.yaml
+           :language: yaml
+           :linenos:
+
+
+Editing the template
+--------------------
+
+Now that you have a ``.readthedocs.yaml`` file added to your Git repository,
+you should see Read the Docs trying to build your project with the configuration file.
```

Comment:
We actually don't make it obvious to users that the build is using a configuration file. We do surface this in the beta dashboard via the debug command output, but that's also the effective configuration file, not the actual file. It might not be clear to the user that the build is doing anything different.

We've talked a number of times of adding a debug flag to commands so we could output this level of information to users without it being noise, but that is still a ways out and probably only a beta dashboard thing still.

I'm not sure exactly what to say here though, other than "you should see the build doing something different" :thinking:

Reply:
OH! Great point!! This is actually _a problem_. I just looked at a build, and no not even the raw log has this info. Can't we fix that in a separate issue very quickly? I'm not sure what's the best approach, but I imagine that this shouldn't have astronomic complexity? :)

Reply:
Oh haha well, interesting timing you had for that input: https://github.com/readthedocs/readthedocs.org/pull/10413

Reply:
Since this PR is already open, I think we can resolve this part of the how-to and try to align these two efforts.

Reply:
Yeah, that makes sense, this is indeed a problem we should resolve. I think it's fine to mention this in our docs even if our UI doesn't exactly match.